### PR TITLE
get current roam node content when dialog opens

### PR DESCRIPTION
**Pull request by mdroidian**
https://www.loom.com/share/cf6b07dcce304f59b5a7425ce383724b

### CODE REVIEW

File: src/components/TldrawCanvasLabelDialog.tsx [hunk 1]
```diff
@@ -306,15 +308,19 @@ const LabelDialog = ({
   const containerRef = useRef<HTMLDivElement>(null);
   const [error, setError] = useState("");
   const initialLabel = useMemo(() => {
-    if (_label) return _label;
+    if (_label)
+      return getPageTitleByPageUid(initialUid) || getTextByBlockUid(initialUid);
```
Comment (dvargas92495) [inline src/components/TldrawCanvasLabelDialog.tsx:312]: why does truthiness of `_label` dictate to use data from initialUid instead?

File: src/components/TldrawCanvasLabelDialog.tsx [hunk 2]
```diff
@@ -306,15 +308,19 @@ const LabelDialog = ({
...
     return getPlainTitleFromSpecification({ specification, text });
-  }, [_label, nodeType]);
+  }, [_label, nodeType, initialUid, isOpen]);
   const initialValue = useMemo(() => {
     return { text: initialLabel, uid: initialUid };
   }, [initialLabel, initialUid]);
-  const [label, setLabel] = useState(initialValue.text);
+  const [label, setLabel] = useState("");
+  useEffect(() => {
+    setLabel(initialLabel);
```
Comment (dvargas92495) [inline src/components/TldrawCanvasLabelDialog.tsx:322]: I think bc `_label` updates, `initialLabel` updates, so doesn't this fire on every keystroke?

File: src/components/TldrawCanvasLabelDialog.tsx (hunk 1 above)
Comment (mdroidian) [reply src/components/TldrawCanvasLabelDialog.tsx]: it's a proxy for "Is this a new canvas node or an existing canvas node?"

If it is existing, get most up to date node content.
If it's new, get the `specification` text instead

It's not used elsewhere so I renamed it to `isExistingCanvasNode`

File: src/components/TldrawCanvasLabelDialog.tsx (hunk 2 above)
Comment (mdroidian) [reply src/components/TldrawCanvasLabelDialog.tsx]: I think `_label` only updated when `shape.props.title` changed, which is `onSuccess` of the dialog.

https://www.loom.com/share/09f71b10c61b4dbaa8aad2bbb61f0308

Regardless, testing this showed that it was firing for every node on canvas load.  Changed it to fire on `isOpen`

**Review by dvargas92495 — APPROVED** [review]

**Pull request merged**